workspaceitem-data-license.md

```diff
@@ -0,0 +1,171 @@
+# WorkspaceItem data of license sectionType
+[Back to the definition of the workspaceitems endpoint](workspaceitems.md)
+
+The section data represent the data about the granted deposit license
+
+```json
+{
+  	"url": "https://dspace7.4science.it/dspace-spring-rest/api/core/bitstreams/004a297e-fd06-4662-ae51-73e4b7c165c8/content",
+    "acceptanceDate": "2017-11-20T10:32:42Z"
+}
+```
+
+* url (**READ-ONLY**): the URL where the accepted license is visible
+* acceptanceDate: the timestamp of the acceptance as recorded in the dcterms:dataAccepted metadata of the license bitstream. For license accepted before than DSpace7 the acceptanceDate will be *null* 
+
+## Patch operations
+The PATCH method expects a JSON body according to the [JSON Patch specification RFC6902](https://tools.ietf.org/html/rfc6902)
+
+Each successful Patch operation will return a HTTP 200 CODE with the new workspaceitem as body 
+
+### Add
+To accept a license the timestamp of the acceptance the client must send a JSON Patch ADD operation as follow
+
+`curl --data '{[ { "op": "add", "path": "/sections/<:name-of-the-form>/acceptanceDate", "value": "YYYY-MM-DDTHH:MM:SSZ"}]}' -X PATCH ${dspace7-url}/api/submission/workspaceitems/<:id>`
```

Comment:
Doesn't this enable a client to cheat with his acceptance date (e.g. take the date of last month, or two months in the future)? I give an alternative below.

Reply:
Yes and not. We can implement validation on server side and avoid excessive difference from the local time. Storing the exact time when the user has grant the license could be necessary to produce a license that match exactly what the user has seen (if you put a timestamp in the license text using placeholder variable https://wiki.duraspace.org/display/DSDOC6x/Configuration+Reference#ConfigurationReference-SubmissionLicenseSubstitutionVariables ). If we don't feel that the previous argument is strong enough, we can also decide to use the timestamp only as guide value (to check that the template was still valid) and store in the system the server time

Reply:
we can also use an additional, separate metadata, to store the server time anyway